sdk/go/pulumi/context.go

```diff
@@ -1182,10 +1182,10 @@ func (ctx *Context) makeResourceState(t, name string, resourceV Resource, provid
 
 	// Populate ResourceState resolvers. (Pulled into function to keep the nil-ness linter check happy).
 	populateResourceStateResolvers := func() {
-		contract.Assert(rs != nil)
+		contract.Assertf(rs != nil, "ResourceState must not be nil")
 		if rs.urn.OutputState != nil {
 			err := ctx.Log.Error(fmt.Sprintf("The resource named %v (type: %v) was initialized multiple times.", name, t), nil)
-			contract.AssertNoError(err)
+			contract.AssertNoErrorf(err, "failed to log error")
```

Comment:
Wonder if this should actually be an IgnoreError? I don't think we should ever panic just because logging failed.

Reply:
Good thought. Updated to IgnoreError for both log errors cases.